src/Microsoft.DotNet.ProjectJsonMigration/ProjectDependencyFinder.cs

```diff
@@ -104,10 +105,14 @@ public IEnumerable<ProjectDependency> ResolveProjectDependencies(ProjectContext
                 var projectExportName = projectExport.Library.Identity.Name;
                 ProjectDependency projectDependency;
 
+                if (preResolvedProjects.Contains(projectExportName))
```

Comment:
Had a chat with @brthor decided to leave this piece of code since csproj reference takes preference over project.json reference with the same name.